File: app/src/main/java/com/example/projectrumba/components/DeviceAdapter.java
package com.example.projectrumba.components;

import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.Spinner;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.fragment.app.FragmentManager;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class DeviceAdapter extends RecyclerView.Adapter<DeviceAdapter.DeviceViewHolder>{

    private List<String> deviceList;
    private View.OnClickListener onClickListener;

    public DeviceAdapter(List<String> deviceList, View.OnClickListener onClickListener) {
        this.deviceList = deviceList;
        this.onClickListener = onClickListener;
    }
    @NonNull
    @Override
    public DeviceViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(android.R.layout.simple_list_item_1, parent, false);
        return new DeviceViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull DeviceViewHolder holder, int position) {
        String deviceName = deviceList.get(position);
        holder.textView.setText(deviceName);
        holder.itemView.setTag(deviceName);
        holder.itemView.setOnClickListener(onClickListener);
    }

    @Override
    public int getItemCount() {
        return deviceList.size();
    }

    public static class DeviceViewHolder extends RecyclerView.ViewHolder {
        TextView textView;

        public DeviceViewHolder(@NonNull View itemView) {
            super(itemView);
            textView = itemView.findViewById(android.R.id.text1);
        }
    }
}

File: app/src/main/java/com/example/projectrumba/components/DeviceAdapter.kt
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.projectrumba.R

class DeviceAdapter (private val ipAddresses: List<String>) : RecyclerView.Adapter<DeviceAdapter.IPViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): IPViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_device_near, parent, false)
        return IPViewHolder(view)
    }

    override fun onBindViewHolder(holder: IPViewHolder, position: Int) {
        val addresses = ipAddresses[position]
        holder.ipTextView.text = addresses.toString();
    }

    override fun getItemCount(): Int {
        return ipAddresses.size
    }

    class IPViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val ipTextView: TextView = view.findViewById(R.id.ipText)
    }
}
